apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/GraphQLCompiler.kt

```diff
@@ -119,6 +122,7 @@ class GraphQLCompiler {
       val ir: CodeGenerationIR,
       val outputDir: File,
       val customTypeMap: Map<String, String>,
+      val customIdGenerator: CustomIdGenerator? = null,
```

Comment:
nit: should we make it as non optional? With default value `Sha256IdGenerator `

Reply:
I did try this before. Looks like, in Kotlin, the optional parameters are a compile time thing. So, like in tests, when you have 

```kt
val operationIdGenerator = when (folder.name) { 
  // ... 
  else -> null
}

GraphQLCompiler.Arguments(
  operationIdGenerator = operationIdGenerator,
)
```

the null doesn't cast to the missing parameter and there is no value to represent a missing parameter because it's a compile time thing. There were workarounds like a secondary constructor or using `if (x == null) Arguments() else Arguments(p = p)`. Can you suggest something regarding this?

Reply:
You could just do:

```kotlin
      val customIdGenerator = when (folder.name) {
        "custom_id_generator" -> object : CustomIdGenerator {
          override fun apply(queryString: String, queryFilepath: String): String {
            return "hash"
          }

          override val version: String = "1"
        }
        else -> OperationIdGenerator.Sha256
      }
```

Reply:
The default generator (`OperationIdGenerator.Sha256`) is now specified in 3 places (not including tests) - 

1. Arguments, 
1. plugin (arguments construction - default instead of null
1. plugin-incubating (arguments construction - default instead of null)